app/uk/gov/hmrc/transitmovementsvalidator/routing/VersionedRoutingController.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovementsvalidator.routing
+
+import org.apache.pekko.stream.Materializer
+import org.apache.pekko.stream.scaladsl.Source
+import org.apache.pekko.util.ByteString
+import play.api.mvc.Action
+import play.api.mvc.ControllerComponents
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+import uk.gov.hmrc.transitmovementsvalidator.config.AppConfig
+import uk.gov.hmrc.transitmovementsvalidator.controllers.{MessagesController => TransitionalMessagesController}
+import uk.gov.hmrc.transitmovementsvalidator.models.Final
+import uk.gov.hmrc.transitmovementsvalidator.models.Transitional
+import uk.gov.hmrc.transitmovementsvalidator.stream.StreamingParsers
+import uk.gov.hmrc.transitmovementsvalidator.v2_1.controllers.{MessagesController => FinalMessagesController}
+
+import javax.inject.Inject
+
+class VersionedRoutingController @Inject() (
```

Comment:
Please include the unit test cases for this newly created file

Reply:
This has been done via each of the messages controllers, I was following a similar convention to other routing files in the project

Reply:
Please see here, how it's written for routing : https://github.com/hmrc/common-transit-convention-traders/blob/5c59c431183797ba03e6718c2aba89819f1060a8/test/routing/DeparturesRouterSpec.scala

Reply:
There is no test for `ContentTypeRouting` in this repository, if you would prefer for me to have it as a separate file that's fine, i'll work on it :)

Reply:
Yes please, our test should route to either transition or final based on APIVersion header value we receive from common-transit-convention-traders.

Reply:
Yes, those tests are there just not in a separate file, but that's fine :)